pkg/asset/ignition/bootstrap/baremetal/template.go

```diff
@@ -111,14 +218,47 @@ func GetTemplateData(config *baremetal.Platform, networks []types.MachineNetwork
 	templateData.ExternalStaticDNS = config.BootstrapExternalStaticDNS
 	templateData.ExternalMACAddress = config.ExternalMACAddress
 
-	_, externalURLv6 := externalURLs(config.APIVIPs)
+	// If the user has manually set disableVirtualMediaTLS to False in the Provisioning CR, then enable TLS in ironic.
+	// The default value of 'true' is temporary (for backporting to older releases), which will be changed to 'false' in the future.
+	templateData.DisableIronicVirtualMediaTLS = true
+	protocol := "http"
 
+	type provisioningCRTemplate struct {
+		Spec struct {
+			DisableVirtualMediaTLS *bool `yaml:"disableVirtualMediaTLS"`
+		} `yaml:"spec"`
+	}
+	provisioningCR := &provisioningCRTemplate{}
+	openshiftManifests := &manifests.Openshift{}
+	dependencies.Get(openshiftManifests)
+	var provisioningCRBytes []byte
+	for _, file := range openshiftManifests.Files() {
+		if strings.Contains(file.Filename, "99_baremetal-provisioning-config.yaml") {
+			logrus.Debugf("Provisioning CR found")
+			provisioningCRBytes = file.Data
+			break
+		}
+	}
+	if provisioningCRBytes != nil {
+		err := yaml.Unmarshal(provisioningCRBytes, provisioningCR)
+		if err != nil {
+			logrus.Debugf("Error in unmarshalling Provisioning CR while generating TLS certificate for ironic virtual media: %s", err)
+		}
+	}
+	if provisioningCR.Spec.DisableVirtualMediaTLS != nil {
+		templateData.DisableIronicVirtualMediaTLS = *provisioningCR.Spec.DisableVirtualMediaTLS
+		if !*provisioningCR.Spec.DisableVirtualMediaTLS {
+			logrus.Debugf("TLS is enabled for ironic virtual media")
+			protocol = "https"
+		}
+	}
+	_, externalURLv6 := externalURLs(config.APIVIPs, protocol)
 	templateData.ExternalURLv6 = externalURLv6
 
 	if len(config.APIVIPs) > 0 {
 		templateData.APIVIPs = config.APIVIPs
-		templateData.BaremetalEndpointOverride = fmt.Sprintf("http://%s/v1", net.JoinHostPort(config.APIVIPs[0], "6385"))
-		templateData.BaremetalIntrospectionEndpointOverride = fmt.Sprintf("http://%s/v1", net.JoinHostPort(config.APIVIPs[0], "5050"))
+		templateData.BaremetalEndpointOverride = fmt.Sprintf("%s://%s/v1", protocol, net.JoinHostPort(config.APIVIPs[0], "6385"))
```

Comment:
Oh. These two are for Ironic (and removed Inspector at all), we should not change them (for now). Only external URLs.

Reply:
Fixed